Type 'unknown' is not assignable to ...

### Description

Compiling the generated api client fails on multiple occasions

```
generated/sdk.gen.ts:1090:82 - error TS2345: Argument of type '{ headers: { 'Content-Type': string; } | { length: number; toString(): string; toLocaleString(): string; pop(): [string, string] | undefined; push(...items: [string, string][]): number; ... 36 more ...; 'Content-Type': string; } | { ...; } | { ...; } | { ...; }; ... 26 more ...; url: string; }' is not assignable to parameter of type 'Omit<RequestOptions<ThrowOnError, string>, "method">'.
  Types of property 'body' are incompatible.
    Type 'unknown' is not assignable to type 'number | unknown[] | Record<string, unknown> | BodyInit | Record<string, unknown>[] | null | undefined'.

1090     return (options.client ?? _heyApiClient).put<unknown, unknown, ThrowOnError>({
                                                                                      ~
1091         security: [
     ~~~~~~~~~~~~~~~~~~~
 ... 
1106         }
     ~~~~~~~~~
1107     });
     ~~~~~

generated/sdk.gen.ts:1793:104 - error TS2345: Argument of type '{ headers: { 'Content-Type': string; } | { length: number; toString(): string; toLocaleString(): string; pop(): [string, string] | undefined; push(...items: [string, string][]): number; ... 36 more ...; 'Content-Type': string; } | { ...; } | { ...; } | { ...; }; ... 26 more ...; url: string; }' is not assignable to parameter of type 'Omit<RequestOptions<ThrowOnError, string>, "method">'.
  Types of property 'body' are incompatible.
    Type 'unknown' is not assignable to type 'number | unknown[] | Record<string, unknown> | BodyInit | Record<string, unknown>[] | null | undefined'.

1793     return (options.client ?? _heyApiClient).put<unknown, SetDashboardItemPropertyError, ThrowOnError>({
                                                                                                            ~
1794         security: [
     ~~~~~~~~~~~~~~~~~~~
 ...
1809         }
     ~~~~~~~~~
1810     });
     ~~~~~

generated/sdk.gen.ts:5695:82 - error TS2345: Argument of type '{ headers: { 'Content-Type': string; } | { length: number; toString(): string; toLocaleString(): string; pop(): [string, string] | undefined; push(...items: [string, string][]): number; ... 36 more ...; 'Content-Type': string; } | { ...; } | { ...; } | { ...; }; ... 26 more ...; url: string; }' is not assignable to parameter of type 'Omit<RequestOptions<ThrowOnError, string>, "method">'.
  Types of property 'body' are incompatible.
    Type 'unknown' is not assignable to type 'number | unknown[] | Record<string, unknown> | BodyInit | Record<string, unknown>[] | null | undefined'.

5695     return (options.client ?? _heyApiClient).put<unknown, unknown, ThrowOnError>({
                                                                                      ~
5696         security: [
     ~~~~~~~~~~~~~~~~~~~
 ...
5711         }
     ~~~~~~~~~
5712     });
     ~~~~~

generated/sdk.gen.ts:6499:82 - error TS2345: Argument of type '{ headers: { 'Content-Type': string; } | { length: number; toString(): string; toLocaleString(): string; pop(): [string, string] | undefined; push(...items: [string, string][]): number; ... 36 more ...; 'Content-Type': string; } | { ...; } | { ...; } | { ...; }; ... 26 more ...; url: string; }' is not assignable to parameter of type 'Omit<RequestOptions<ThrowOnError, string>, "method">'.
  Types of property 'body' are incompatible.
    Type 'unknown' is not assignable to type 'number | unknown[] | Record<string, unknown> | BodyInit | Record<string, unknown>[] | null | undefined'.

6499     return (options.client ?? _heyApiClient).put<unknown, unknown, ThrowOnError>({
                                                                                      ~
6500         security: [
     ~~~~~~~~~~~~~~~~~~~
 ...
6515         }
     ~~~~~~~~~
6516     });
     ~~~~~

generated/sdk.gen.ts:7427:105 - error TS2345: Argument of type '{ headers: { 'Content-Type': string; } | { length: number; toString(): string; toLocaleString(): string; pop(): [string, string] | undefined; push(...items: [string, string][]): number; ... 36 more ...; 'Content-Type': string; } | { ...; } | { ...; } | { ...; }; ... 26 more ...; url: string; }' is not assignable to parameter of type 'Omit<RequestOptions<ThrowOnError, string>, "method">'.
  Types of property 'body' are incompatible.
    Type 'unknown' is not assignable to type 'number | unknown[] | Record<string, unknown> | BodyInit | Record<string, unknown>[] | null | undefined'.

7427     return (options.client ?? _heyApiClient).post<CreateIssueTypeAvatarResponse, unknown, ThrowOnError>({
                                                                                                             ~
7428         security: [
     ~~~~~~~~~~~~~~~~~~~
 ...
7443         }
     ~~~~~~~~~
7444     });
     ~~~~~

generated/sdk.gen.ts:7535:82 - error TS2345: Argument of type '{ headers: { 'Content-Type': string; } | { length: number; toString(): string; toLocaleString(): string; pop(): [string, string] | undefined; push(...items: [string, string][]): number; ... 36 more ...; 'Content-Type': string; } | { ...; } | { ...; } | { ...; }; ... 26 more ...; url: string; }' is not assignable to parameter of type 'Omit<RequestOptions<ThrowOnError, string>, "method">'.
  Types of property 'body' are incompatible.
    Type 'unknown' is not assignable to type 'number | unknown[] | Record<string, unknown> | BodyInit | Record<string, unknown>[] | null | undefined'.

7535     return (options.client ?? _heyApiClient).put<unknown, unknown, ThrowOnError>({
                                                                                      ~
7536         security: [
     ~~~~~~~~~~~~~~~~~~~
 ...
7551         }
     ~~~~~~~~~
7552     });
     ~~~~~

generated/sdk.gen.ts:10624:103 - error TS2345: Argument of type '{ headers: { 'Content-Type': string; } | { length: number; toString(): string; toLocaleString(): string; pop(): [string, string] | undefined; push(...items: [string, string][]): number; ... 36 more ...; 'Content-Type': string; } | { ...; } | { ...; } | { ...; }; ... 26 more ...; url: string; }' is not assignable to parameter of type 'Omit<RequestOptions<ThrowOnError, string>, "method">'.
  Types of property 'body' are incompatible.
    Type 'unknown' is not assignable to type 'number | unknown[] | Record<string, unknown> | BodyInit | Record<string, unknown>[] | null | undefined'.

10624     return (options.client ?? _heyApiClient).post<CreateProjectAvatarResponse, unknown, ThrowOnError>({
                                                                                                            ~
10625         security: [
      ~~~~~~~~~~~~~~~~~~~
  ...
10640         }
      ~~~~~~~~~
10641     });
      ~~~~~

generated/sdk.gen.ts:10966:82 - error TS2345: Argument of type '{ headers: { 'Content-Type': string; } | { length: number; toString(): string; toLocaleString(): string; pop(): [string, string] | undefined; push(...items: [string, string][]): number; ... 36 more ...; 'Content-Type': string; } | { ...; } | { ...; } | { ...; }; ... 26 more ...; url: string; }' is not assignable to parameter of type 'Omit<RequestOptions<ThrowOnError, string>, "method">'.
  Types of property 'body' are incompatible.
    Type 'unknown' is not assignable to type 'number | unknown[] | Record<string, unknown> | BodyInit | Record<string, unknown>[] | null | undefined'.

10966     return (options.client ?? _heyApiClient).put<unknown, unknown, ThrowOnError>({
                                                                                       ~
10967         security: [
      ~~~~~~~~~~~~~~~~~~~
  ...
10982         }
      ~~~~~~~~~
10983     });
      ~~~~~

generated/sdk.gen.ts:13430:95 - error TS2345: Argument of type '{ headers: { 'Content-Type': string; } | { length: number; toString(): string; toLocaleString(): string; pop(): [string, string] | undefined; push(...items: [string, string][]): number; ... 36 more ...; 'Content-Type': string; } | { ...; } | { ...; } | { ...; }; ... 26 more ...; url: string; }' is not assignable to parameter of type 'Omit<RequestOptions<ThrowOnError, string>, "method">'.
  Types of property 'body' are incompatible.
    Type 'unknown' is not assignable to type 'number | unknown[] | Record<string, unknown> | BodyInit | Record<string, unknown>[] | null | undefined'.

13430     return (options.client ?? _heyApiClient).post<StoreAvatarResponse, unknown, ThrowOnError>({
                                                                                                    ~
13431         security: [
      ~~~~~~~~~~~~~~~~~~~
  ...
13446         }
      ~~~~~~~~~
13447     });
      ~~~~~

generated/sdk.gen.ts:14032:82 - error TS2345: Argument of type '{ headers: { 'Content-Type': string; } | { length: number; toString(): string; toLocaleString(): string; pop(): [string, string] | undefined; push(...items: [string, string][]): number; ... 36 more ...; 'Content-Type': string; } | { ...; } | { ...; } | { ...; }; ... 26 more ...; url: string; }' is not assignable to parameter of type 'Omit<RequestOptions<ThrowOnError, string>, "method">'.
  Types of property 'body' are incompatible.
    Type 'unknown' is not assignable to type 'number | unknown[] | Record<string, unknown> | BodyInit | Record<string, unknown>[] | null | undefined'.

14032     return (options.client ?? _heyApiClient).put<unknown, unknown, ThrowOnError>({
                                                                                       ~
14033         security: [
      ~~~~~~~~~~~~~~~~~~~
  ...
14048         }
      ~~~~~~~~~
14049     });
      ~~~~~

generated/sdk.gen.ts:17613:165 - error TS2345: Argument of type '{ headers: { 'Content-Type': string; } | { length: number; toString(): string; toLocaleString(): string; pop(): [string, string] | undefined; push(...items: [string, string][]): number; ... 36 more ...; 'Content-Type': string; } | { ...; } | { ...; } | { ...; }; ... 26 more ...; url: string; }' is not assignable to parameter of type 'Omit<RequestOptions<ThrowOnError, string>, "method">'.
  Types of property 'body' are incompatible.
    Type 'unknown' is not assignable to type 'number | unknown[] | Record<string, unknown> | BodyInit | Record<string, unknown>[] | null | undefined'.

17613     return (options.client ?? _heyApiClient).put<AddonPropertiesResourcePutAddonPropertyPutResponse, AddonPropertiesResourcePutAddonPropertyPutError, ThrowOnError>({
                                                                                                                                                                          ~
17614         security: [
      ~~~~~~~~~~~~~~~~~~~
  ...
17629         }
      ~~~~~~~~~
17630     });
      ~~~~~

generated/sdk.gen.ts:17778:119 - error TS2345: Argument of type '{ headers: { 'Content-Type': string; } | { length: number; toString(): string; toLocaleString(): string; pop(): [string, string] | undefined; push(...items: [string, string][]): number; ... 36 more ...; 'Content-Type': string; } | { ...; } | { ...; } | { ...; }; ... 26 more ...; url: string; }' is not assignable to parameter of type 'Omit<RequestOptions<ThrowOnError, string>, "method">'.
  Types of property 'body' are incompatible.
    Type 'unknown' is not assignable to type 'number | unknown[] | Record<string, unknown> | BodyInit | Record<string, unknown>[] | null | undefined'.

17778     return (options.client ?? _heyApiClient).put<PutForgeAppPropertyResponse, PutForgeAppPropertyError, ThrowOnError>({
                                                                                                                            ~
17779         security: [
      ~~~~~~~~~~~~~~~~~~~
  ...
17794         }
      ~~~~~~~~~
17795     });
      ~~~~~


Found 12 errors in the same file, starting at: generated/sdk.gen.ts:1090
```

### Reproducible example or configuration

- use this `openapi-ts.config.ts` file
```ts
import { defineConfig } from '@hey-api/openapi-ts';

export default defineConfig({
  input: 'swagger.json',
  plugins: [
    '@hey-api/client-fetch',
    '@hey-api/sdk',
    '@hey-api/typescript',
    '@hey-api/transformers',
  ],
  output: {
    format: 'prettier',
    lint: 'eslint',
    path: 'generated',
  },
});
```
- generate client with `openapi-ts`
- use this `tsconfig.json`
```json
{
  "compilerOptions": {
    "module": "commonjs",
    "lib": ["esnext", "dom"],
    "target": "esnext",
    "esModuleInterop": false,
    "isolatedModules": false,
    "experimentalDecorators": true,
    "emitDecoratorMetadata": true,
    "strict": true,
    "strictNullChecks": true,
    "outDir": "./dist"
  },
  "exclude": ["dist", "openapi-ts.config.ts", "node_modules"]
}
```
- compile with `tsc -d --declarationMap`

### OpenAPI specification (optional)

https://dac-static.atlassian.com/cloud/jira/platform/swagger-v3.v3.json?_v=1.7686.0-0.1317.0

### System information (optional)

- `@hey-api/openapi-ts@0.63.2`
- `@hey-api/client-fetch@0.7.3`
- `typescript@5.4.5`

--- a/packages/client-fetch/src/types.ts
+++ b/packages/client-fetch/src/types.ts
@@ -48,12 +48,7 @@ export interface RequestOptions<
    *
    * {@link https://developer.mozilla.org/docs/Web/API/fetch#body}
    */
-  body?:
-    | RequestInit['body']
-    | Record<string, unknown>
-    | Array<Record<string, unknown>>
-    | Array<unknown>
-    | number;
+  body?: unknown;
   path?: Record<string, unknown>;
   query?: Record<string, unknown>;
   /**

--- a/packages/client-next/src/types.ts
+++ b/packages/client-next/src/types.ts
@@ -48,12 +48,7 @@ export interface RequestOptions<
    *
    * {@link https://developer.mozilla.org/docs/Web/API/fetch#body}
    */
-  body?:
-    | RequestInit['body']
-    | Record<string, unknown>
-    | Array<Record<string, unknown>>
-    | Array<unknown>
-    | number;
+  body?: unknown;
   path?: Record<string, unknown>;
   query?: Record<string, unknown>;
   /**

--- a/packages/client-nuxt/src/types.ts
+++ b/packages/client-nuxt/src/types.ts
@@ -68,7 +68,7 @@ export interface RequestOptions<
        *
        * {@link https://developer.mozilla.org/docs/Web/API/fetch#body}
        */
-      body?: BodyInit | Record<string, any> | null;
+      body?: unknown;
       path?: FetchOptions<unknown>['query'];
       query?: FetchOptions<unknown>['query'];
     }> {

